### PR TITLE
zsh tutorial v1

### PR DESCRIPTION
**Pull request by nickysemenza**
first iteration of it - not sure what other main sections i should add

### CODE REVIEW

**Comment by scottopell** [comment]
Since you started to talk about oh-my-zsh, I would say that adding a section on how to set it up may be worthwhile, or at least link to the official install guide.

Under "Scripting", nothing that you use there is zsh specific, so we should break that out into its own page and only use POSIX-compliant scripts in that one, then in here we can mention some of the niceties that zsh scripting allows for.

One other thing to do would be to talk about themes, that's huge with people, some people like to use the gaudiest themes.

And lastly I would say a big section should be dedicated to getting utf8 support in each terminal.

I wrote up a guide on how to do in on the lab machines somewhere in last year's piazza and this was a big oversight on my part last year when we taught it. So many people ended up with the question mark boxes in their terminal because we never covered this at all.


On a slightly related note, somewhere we should document the definitions of shell, terminal, posix(and what it means in terms of shells), bash, zsh, etc and what they're all used for. TA's should try to use the most accurate terminology whenever possible, because they just take whatever we say and run with it.

File: tutorials/shells/zsh.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#ZSH
+ZSH is one of many alternative shells. Most machines that you will encounter will come with bash installed.
+
+ZSH makes it very easy to use plugins, make UI tweaks, and increase productivity in general
+
+here is an example of what my terminal looks like when i run `ls`
+![](zsh_resources/zsh-ls.png =450x)
```
Comment (scottopell) [inline tutorials/shells/zsh.md]: Is this valid github flavored markdown? It doesn't render for me

File: tutorials/shells/zsh.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+###Scripting
+You add scripts to your `.zshrc` to make your life easier
+Here are some I have in mine:
+####Compile and run a java file
+Usage: **jav [filename]**
+where filename.java is your file
+
+```
+jav() {
+	javac "$1.java";
```
Comment (scottopell) [inline tutorials/shells/zsh.md]: This may  move, but wherever it ends up, we should go over what this magical `$1` is.

**Comment by leviathan747** [comment]
Good start. I agree with all @scottopell 's comments. Also, maybe just a little bit more explanation in each section. I know it seems really obvious/redundant, but many of these students are being introduced to the concept of a shell for the very first time.

As to the discussion about the scripting section, I agree with Scott that it should be done in a more general sense. I would recommend either a separate tutorial on shells in general or include it in the bash tutorial (since bash is quite standard).

@scottopell , I included a lot more explanation of the terms that you mentioned in the first lecture on shells. I think last year we jumped into commands too fast and didn't give them enough time to grasp WTF they were even using.

**Comment by scottopell** [comment]
@leviathan747 sounds like a good improvement! My only comment would be that it's important to get tangible commands out sooner rather than later so that they know it's useful. They can go a long time and we probably have a lot of juniors and seniors that use a shell every day without knowing exactly what a shell is.

But I agree more knowledge is definitely better, especially if people felt lost or disoriented

**Comment by nickysemenza** [comment]
@leviathan747 I agree, it could make more sense to refactor the structure to have a general document about shells & scripting, and then have some mini components about some of the different popular shells and what they have to offer

**Comment by jay-hankins** [comment]
@nickysemenza Would you like to finish up refactoring this PR?

**Pull request closed without merging**